template.rb

```diff
@@ -193,59 +201,41 @@ def create_database_and_initial_migration
 end
 
 def add_yarn_start_script
+  return add_package_json_script(start: "bin/dev") if File.exist?("bin/dev")
+
   run_with_clean_bundler_env "yarn add --dev concurrently"
-  add_package_json_script(
-    start: "concurrently --raw --kill-others-on-fail 'bin/rails s -b 0.0.0.0' 'bin/webpack-dev-server' 'bin/sidekiq'"
-  )
+
+  procs = ["'bin/rails s -b 0.0.0.0'", "bin/sidekiq"]
+  procs << "'bin/webpack-dev-server'" if File.exist?("bin/webpack-dev-server")
+
+  add_package_json_script(start: "concurrently --raw --kill-others-on-fail #{procs.join(" ")}")
 end
 
 def add_yarn_lint_and_run_fix
   packages = %w[
-    babel-eslint
-    eslint@^7.32.0
+    eslint
     eslint-config-prettier
-    eslint-plugin-prettier prettier
+    eslint-plugin-prettier
     npm-run-all
-    stylelint@^13.13.1
-    stylelint-config-recommended-scss@^4.3.0
-    stylelint-config-standard@^22.0.0
+    prettier
+    stylelint
+    stylelint-config-recommended-scss
+    stylelint-config-standard
     stylelint-declaration-use-variable
-    stylelint-scss@^3.21.0
+    stylelint-scss
   ]
   run_with_clean_bundler_env "yarn add #{packages.join(' ')} -D"
   add_package_json_script("lint": "npm-run-all -c lint:*")
-  add_package_json_script("lint:js": "eslint 'app/{assets,components,javascript}/**/*.{js,jsx}'")
-  add_package_json_script("lint:css": "stylelint 'app/{assets,components,javascript}/**/*.{css,scss}'")
+  add_package_json_script("lint:js": "eslint 'app/{components,frontend,javascript}/**/*.{js,jsx}'")
+  add_package_json_script("lint:css": "stylelint 'app/{components,frontend,assets/stylesheets}/**/*.{css,scss}'")
   run_with_clean_bundler_env "yarn lint:js --fix"
   run_with_clean_bundler_env "yarn lint:css --fix"
 end
 
 def add_package_json_script(scripts)
-  package_json = JSON.parse(IO.read("package.json"))
-  package_json["scripts"] ||= {}
   scripts.each do |name, script|
-    package_json["scripts"][name.to_s] = script
+    run ["npm", "set-script", name.to_s.shellescape, script.shellescape].join(" ")
```

Comment:
It is much easier to use `npm set-script` rather than manually rewriting `package.json` ourselves in Ruby. 🙂